### PR TITLE
fix: flight examples

### PR DESCRIPTION
**Pull request by codonnell**
## Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/9334.

## Rationale for this change

It allows the arrow flight examples to compile and makes all nested examples runnable.

## What changes are included in this PR?

This PR downgrades the `tonic` dependency to `0.10` to match `arrow-flight` and adds explicit `[[example]]` blocks so nested examples can be run with `cargo run --example`.

## Are these changes tested?

I verified locally that the arrow flight examples compile with this change, and I was able to run queries with JDBC against the `arrow_flight_sql` example.

## Are there any user-facing changes?

No, just example fixes.

### CODE REVIEW

File: datafusion-examples/Cargo.toml [hunk 1]
```diff
@@ -54,6 +78,6 @@ serde = { version = "1.0.136", features = ["derive"] }
 serde_json = { workspace = true }
 tempfile = { workspace = true }
 tokio = { workspace = true, features = ["rt-multi-thread", "parking_lot"] }
-tonic = "0.11"
```
Comment (comphead) [inline datafusion-examples/Cargo.toml:57]: is this change related?

Comment (codonnell) [reply datafusion-examples/Cargo.toml:57]: I tried to capture this in the issue:
> There is an API incompatibility between tonic 0.10 and 0.11. We have 0.10 as a transitive dependency of arrow-flight, and datafusion-examples pulls in 0.11 directly. As a result the flight examples won't compile.
If that's not clear, I'm happy to clarify further.

**Review by Jefffrey — APPROVED** [review]
Makes sense, thanks for this

Do we run (or at least build) these examples as part of the CI? Since if so that should've prevented the bump in the first place: https://github.com/apache/arrow-datafusion/pull/9176

arrow-flight will have tonic version `0.11` in it's next release as it was also bumped: https://github.com/apache/arrow-rs/pull/5380

So we can downgrade for now until arrow `51.0.0` is released

**Comment by codonnell** [comment]
> Do we run (or at least build) these examples as part of the CI? Since if so that should've prevented the bump in the first place: #9176

I wondered about that, too. The flight examples were excluded in [this refactor](https://github.com/apache/arrow-datafusion/pull/8060/files#diff-afe5c39a3bee2aaaf614226a287b288aea04d8b914b22ee2c5188ea7d9fb8e90R28), presumably because the server examples run indefinitely and the client example needs a server to connect to. We _could_ add a `cargo check` for each flight sql example to the CI script; I'm not sure why we wouldn't. With this PR, `cargo check` passes for all the examples. I'm happy to add that if you think it would be valuable.

> arrow-flight will have tonic version `0.11` in it's next release as it was also bumped: [apache/arrow-rs#5380](https://github.com/apache/arrow-rs/pull/5380)
> 
> So we can downgrade for now until arrow `51.0.0` is released

My thoughts, exactly. :+1:

**Comment by codonnell** [comment]
Is it worth adding a short comment above the tonic dependency explaining the situation? Might help whoever upgrades arrow understand how to resolve the tonic incompatibility more quickly.

**Comment by codonnell** [comment]
I went ahead and added the CI check and comment. Happy to adjust as needed.

**Review by comphead — APPROVED** [review]
lgtm thanks @codonnell
